app/controllers/reviews_controller.rb

```diff
@@ -6,7 +6,11 @@ def index
   end
 
   def new
-    @companies = current_customer.contracts.map{|contract| contract.quote.company}.uniq
+    if params[:company]
+      @companies = [Company.find_by(id: params[:company])]
+    else
+      @companies = current_customer.contracts.map{|contract| contract.quote.company}.uniq
```

Comment:
Use `current_customer.contracts.includes(:company).map` to avoid n+1 queries. Also, you can call `contract.company` within the block since contract and company have a one_to_many relationship